### PR TITLE
NMS-15753: Allow meta-data and SCV access for VMware credentials

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://opennms.atlassian.net/browse/NMS-15753

### CODE REVIEW

File: docs/modules/reference/pages/service-assurance/monitors/VmwareCimMonitor.adoc [hunk 1]
```diff
@@ -10,6 +10,9 @@ It does not apply to virtual machines.
 This monitor requires hosts be imported with Provisiond and the VMware import.
 {page-component-title} requires network access to VMware vCenter server and each of the monitored hosts.
 The credentials for accessing the status of a VMware entity are collected from the responsible VMware vCenter using the settings from `$OPENNMS_HOME/etc/vmware-config.xml`.
+
+NOTE: Meta-data expressions can also be used in the username and password attributes of the `vmware-config.xml` configuration file. This allows the user to also reference credentials stored in the secure credentials vault.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/service-assurance/monitors/VmwareCimMonitor.adoc:14]:
```suggestion
NOTE: Metadata expressions can also be used in the username and password attributes of the `vmware-config.xml` configuration file. 
This allows the user to also reference credentials stored in the secure credentials vault.
```

File: docs/modules/reference/pages/service-assurance/monitors/VmwareMonitor.adoc [hunk 2]
```diff
@@ -16,6 +16,9 @@ The VmwareMonitor requires that hosts be imported using provisiond and the VMwar
 {page-component-title} requires network access to the VMWare vCenter server and each of the monitored hosts.
 
 The credentials for accessing the status of a VMware entity are collected from the appropriate vCenter instance using settings from `$\{OPENNMS_HOME}/etc/vmware-config.xml`.
+
+NOTE: Meta-data expressions can also be used in the username and password attributes of the `vmware-config.xml` configuration file. This allows the user to also reference credentials stored in the secure credentials vault.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/service-assurance/monitors/VmwareMonitor.adoc:20]:
```suggestion
NOTE: Metadata expressions can also be used in the username and password attributes of the `vmware-config.xml` configuration file. 
This allows the user to also reference credentials stored in the secure credentials vault.
```

**Review by Bonrob2 — COMMENTED** [review]
Two minor docs changes. Thank you!

**Review by synqotik — APPROVED** [review]
Code looks good, but I don't know a lot about this specific functionality.

**Review by RangerRick — APPROVED** [review]

**Review by Bonrob2 — APPROVED** [review]
docs LGTM!

**Comment by christianpape** [comment]
Reworked the handling of attributes for collectors/monitors to not introduce any dependency issues with the secure credentials vault on the minion. The getRuntimeAttributes() can add attributes which are be interpolated outside of the call. This avoid the need to habe the EntityScopeProvider or JCEKSSecureCredentialsVault available in each collector/monitor class.

**Review by RangerRick — APPROVED** [review]
nice, this looks like a clean way to do it

**Pull request merged**